saleor/product/utils.py

```diff
@@ -17,3 +20,39 @@ def products_with_details(user):
                                          'variants__variant_images__image',
                                          'attributes__values')
     return products
+
+
+PricingInfo = namedtuple(
```

Comment:
Sometimes we know the price yet the product is unavailable. Sometimes the product may be in stock be we can't sell it. I feel like this should become `ProductAvailability` and contain some additional fields.